src/main/java/xyz/nucleoid/bedwars/game/config/GeneratorConfig.java

```diff
@@ -0,0 +1,26 @@
+package xyz.nucleoid.bedwars.game.config;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+public record GeneratorConfig(GeneratorLevelConfig level1, GeneratorLevelConfig level2, GeneratorLevelConfig level3, int diamond, long diamondSpawnInterval, int emerald, long emeraldSpawnInterval) {
+    public static final Codec<GeneratorConfig> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            GeneratorLevelConfig.CODEC.fieldOf("level_1").forGetter(GeneratorConfig::level1),
+            GeneratorLevelConfig.CODEC.fieldOf("level_2").forGetter(GeneratorConfig::level2),
+            GeneratorLevelConfig.CODEC.fieldOf("level_3").forGetter(GeneratorConfig::level3),
+            Codec.INT.fieldOf("diamond").forGetter(GeneratorConfig::diamond),
+            Codec.LONG.fieldOf("diamondSpawnInterval").forGetter(GeneratorConfig::diamondSpawnInterval),
```

Comment:
```suggestion
            Codec.LONG.fieldOf("diamond_spawn_interval").forGetter(GeneratorConfig::diamondSpawnInterval),
```
and similar for other fields for consistency